lib/attributor/types/regexp.rb

```diff
@@ -0,0 +1,30 @@
+require_relative '../exceptions'
+
+module Attributor
+  class Regexp
+    include Type
+
+    def self.native_type
+      return ::Regexp
+    end
+
+    def self.load(value, context=Attributor::DEFAULT_ROOT_CONTEXT, **options)
+      unless value.kind_of?(::String) || value.nil?
+        raise IncompatibleTypeError,  context: context, value_type: value.class, type: self
+      end
+
+      value && ::Regexp.new(value, options[:regexp_opts])
```

Comment:
I'm not sure I understand using a special :regexp_opts key passed in.
`.load` is called internally for the normal path, so you'd never be able to pass that in.

Were you thinking of adding opts to the type itself?